### Import Titanic dataset

In [ ]:
import joblib

data = joblib.load( 'preprocessed_data\\Titanic.pkl')

### Split train/test

In [ ]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(data["X_encoded_scaled"], data["y"], test_size=0.33, random_state=42)

### Manually optimise KNeighborsClassifier

In [ ]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score
import pandas as pd

model = KNeighborsClassifier(n_neighbors=4)
model.fit(X_train, y_train)
pred_train = model.predict(X_train)
pred_test = model.predict(X_test)

print("Train ACC: ", accuracy_score(y_train, pred_train))
print("Test ACC: ", accuracy_score(y_test, pred_test))

Train ACC:  0.8763102725366876
Test ACC:  0.7531914893617021


### Manually optimise LogisticRegressionCV

In [ ]:
from sklearn.linear_model import LogisticRegressionCV
from sklearn.metrics import accuracy_score

model = LogisticRegressionCV(Cs=30, cv=5)
model.fit(X_train, y_train)
pred_train = model.predict(X_train)
pred_test = model.predict(X_test)

print("Train ACC: ", accuracy_score(y_train, pred_train))
print("Test ACC: ", accuracy_score(y_test, pred_test))
print(model.coef_)

Train ACC:  0.80083857442348
Test ACC:  0.774468085106383
[[-0.12552717 -0.03885567  0.01571415  0.15786233  0.70765582 -0.70865332
   0.14290085  0.00519813 -0.14909648  0.45604628  0.05222794 -0.50927172]]


### Manually optimise RidgeClassifier

In [ ]:
from sklearn.linear_model import RidgeClassifier
from sklearn.metrics import accuracy_score

model = RidgeClassifier(alpha=10)
model.fit(X_train, y_train)
pred_train = model.predict(X_train)
pred_test = model.predict(X_test)

print("Train ACC: ", accuracy_score(y_train, pred_train))
print("Test ACC: ", accuracy_score(y_test, pred_test))
print(model.coef_)

Train ACC:  0.7966457023060797
Test ACC:  0.7702127659574468
[[-0.47465743 -0.20538754 -0.03944586 -0.03425986  0.4876189  -0.4876189
   0.04190594  0.03060815 -0.07251409  0.41683141 -0.02273264 -0.39409877]]


### Manually optimise LinearSVC

In [ ]:
from sklearn.svm import LinearSVC
from sklearn.metrics import accuracy_score

model = LinearSVC(penalty='l1')
model.fit(X_train, y_train)
pred_train = model.predict(X_train)
pred_test = model.predict(X_test)

print("Train ACC: ", accuracy_score(y_train, pred_train))
print("Test ACC: ", accuracy_score(y_test, pred_test))
print(model.coef_)

Train ACC:  0.80083857442348
Test ACC:  0.7531914893617021
[[-0.97074235 -0.44997749  0.         -0.21974895  1.03481358  0.
   0.026717    0.02645174 -0.06157468  0.63592973  0.         -0.42371164]]


## Best result is:
### LogisticRegressionCV 
### Train ACC:  0.80083857442348
### Test ACC:  0.774468085106383